pandas/tests/io/parser/test_index_col.py

```diff
@@ -222,3 +222,18 @@ def test_index_col_large_csv(all_parsers):
         result = parser.read_csv(path, index_col=[0])
 
     tm.assert_frame_equal(result, df.set_index("a"))
+
+
+def test_index_col_multiindex_columns_no_data(all_parsers):
```

Comment:
can you add all the cases form the op (3 working & the non-working)

Reply:
Done